design/legacy-design/device-service/discovery.md

```diff
@@ -0,0 +1,79 @@
+## Dynamic Device Discovery
+
+### Overview
+
+Some device protocols allow for devices to be discovered automatically. A Device
+Service may include a capability for discovering devices and creating the
+corresponding Device objects within EdgeX. A framework for doing so will be
+implemented in the Device Service SDKs.
+
+The discovery process will operate as follows:
+
+* Discovery is triggered either on an internal timer or by a call to a REST endpoint
+* The SDK will call a function provided by the DS implementation to request a
+device scan
+* The implementation calls back to the SDK with details of devices which it has found
+* The SDK filters these devices against a set of acceptance criteria
+* The SDK adds accepted devices in core-metadata. These are now available in
+the EdgeX system
+
+### Triggering Discovery
+
+A boolean configuration value `Device/Discovery` defaults to false. If this
+value is set true, and the DS implementation supports discovery, discovery is
+enabled.
+
+The SDK will respond to POST requests on the the /discovery endpoint. This call
+will return one of the following codes:
+
+* 200: discovery has been triggered or is already running
```

Comment:
According to the description, I thought `202 Accepted` is a better option here

> 202 Accepted
> The request has been accepted for processing, but the processing has not been completed. The request might or might not be eventually acted upon, and may be disallowed when processing occurs.

Reply:
My recommendation would be `204`

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204

Because the acknowledgement doesn't send any content back and the success code indicates the request was successfully received

Reply:
I am thinking maybe we could return the correlation id for future log tracing of this request.  The whole operation should use the same correlation id to add devices.

Reply:
+1 on @cloudxxx8 suggestion